src/zenml/zen_stores/migrations/versions/03742aa7fdd7_add_secrets.py

```diff
@@ -0,0 +1,51 @@
+"""add_secrets [03742aa7fdd7].
+
+Revision ID: 03742aa7fdd7
+Revises: bea8a6ce3015
+Create Date: 2023-02-09 17:43:34.207799
+
+"""
+import sqlalchemy as sa
+import sqlmodel
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "03742aa7fdd7"
+down_revision = "bea8a6ce3015"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    """Upgrade database schema and/or data, creating a new revision."""
+    op.create_table(
+        "secret",
+        sa.Column("values", sa.TEXT(), nullable=False),
+        sa.Column(
+            "workspace_id", sqlmodel.sql.sqltypes.GUID(), nullable=False
+        ),
+        sa.Column("user_id", sqlmodel.sql.sqltypes.GUID(), nullable=False),
```

Comment:
Secrets can be user-scoped. Leaving those around when a user is deleted is not useful for anything. Particularly for secrets, you don't want sensitive data associated with your account to be left behind after you delete your account.

Reply:
It may make sense to leave the user nullable and set the user to None on creation for workspace scoped secrets, but that is something that breaks a lot of assumptions in the ZenML model, so I'd rather leave it like this until someone actually complains that this is a limitation.